docs/pages/framework/models/upgrades.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Upgrades models in Dojo
+description: Learn how to upgrade models in Dojo.
+---
+
+# Upgrades
+
+In Dojo, all models are upgradeable, which means that when their code changes, they are redeployed 
+to the same address, preserving the existing model storage and therefore the existing model data.
+
+However, to be able to preserve existing model data, there are some limitations.
+
+## General rules
+
+- To be upgradeable, the layout of a model must not be packed (using `IntrospectPacked`).
+
+- For composite data structures like `struct`, `enum`, `tuple` and `array`
+    - they are upgreadable as soon as all their elements are upgreadable,
+    - existing elements cannot be removed, they can only be modified according to the rules described 
+  on this page. New elements can be freely added.
```

Comment:
```suggestion
  on this page.
```
This one is vague since tuple can't have new elements added.
Wdyt about this summary:

- `struct`: new members can be added or their types modified following the rules below in this page. Renaming a field is not permitted.
- `enum`: new variants can be added, but existing ones can't be renamed. The type of a variant can be changed following the rules below.
- `tuple`: types inside the tuple may be changed following the rules below, however the tuple size can't be changed.
- `array`: the inner type of the array can be changed following the rules below.

Reply:
To be more consistent, I changed the PR to allow increasing the size of tuples as it has no impact on existing stored model data.
Knowing that, do you still think this sentence has to be more detailed ? Because the following rules are already described and match with all the composite types:
- element name cannot change,
- element type can only change according to the rules.